### PR TITLE
unit-tests: add tests for block volume expand feature

### PR DESCRIPTION
**Pull request by pkalever**
### What does this PR achieve? Why do we need it?

add tests for block volume expand feature

Signed-off-by: Prasanna Kumar Kalever \<prasanna.kalever@redhat.com\>

### CODE REVIEW

**Comment by pkalever** [comment]
@phlogistonjohn @raghavendra-talur please help me review this PR.

Also, I request your help in identifying the uncovered scenarios if there are any.

File: apps/glusterfs/block_volume_expand_test.go [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+		}
+		return nil
+	})
+
+	e = bve.Exec(app.executor)
+	tests.Assert(t, e == nil, "expected e == nil, got", e)
+
+	// pretend that we have called gluster-block cli and got some info
+	// assumption, gluster-block failed way before expanding the backend file
+	app.xo.MockInfoBlockVolume = func(host string, blockhostingvolume string,
```
Comment (raghavendra-talur) [inline apps/glusterfs/block_volume_expand_test.go:400]: Make this function assignment before calling the exec function. The exec function should then fail. You will look for exec err not nil and then proceed to rollback.

File: apps/glusterfs/block_volume_expand_test.go [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+			tests.Assert(t, e == nil, "expected e == nil, got", e)
+			tests.Assert(t, blockvolume.Info.Size == 100, "expected block volume size == 100, got:", blockvolume.Info.Size)
+		}
+		return nil
+	})
+
+	e = bve.Exec(app.executor)
+	tests.Assert(t, e == nil, "expected e == nil, got", e)
+
+	// pretend that we have called gluster-block cli and got some info
```
Comment (raghavendra-talur) [inline apps/glusterfs/block_volume_expand_test.go:574]: same here.

File: apps/glusterfs/block_volume_expand_test.go (hunk 1 above)
Comment (pkalever) [reply apps/glusterfs/block_volume_expand_test.go:400]: @raghavendra-talur Exec function doesn't use MockInfoBlockVolume, hence it doesn't matter if we assign the function before or after calling Exec()

But I get your point about failing the Exec(), so I will define a fake app.xo.MockBlockVolumeExpand = {} which will return an error, and as a result, Exec() will fail.

**Comment by pkalever** [comment]
> I have made two comments that require change.
> 
> In addition to that, I can suggest one more test case:
> 
> * Have two block volumes, on the same BHV
> * Request for their expansion such that build() for block expansion is run for both of them before either exec runs.
> * It is expected that the second build fails if the first build left less than required space for the second. on the BHV.
> * Then rollback the first build and retry the second build. This time it should succeed.

This make sense to me, I will add this test case and update the PR soon.

Thanks @raghavendra-talur

**Comment by pkalever** [comment]
@raghavendra-talur @phlogistonjohn Updated the PR addressing the review comments from Talur. PTAL.

Thanks!

**Comment by pkalever** [comment]
Thanks @phlogistonjohn, updated the PR by squashing the patches.

**Pull request merged**